src/attribute_value.rs

```diff
@@ -107,6 +107,20 @@ where
     }
 }
 
+impl<T> From<&HashMap<String, T>> for Item
+where
+    AttributeValue: From<T>,
+    T: Clone,
+{
+    fn from(m: &HashMap<String, T>) -> Self {
+        Item(
+            m.into_iter()
+                .map(|(key, value)| (key.clone(), AttributeValue::from(value.clone())))
+                .collect(),
+        )
+    }
+}
```

Comment:
I kinda wonder if this makes more sense to clone the hashmap right away, like the  following. Thoughts?

```rust
impl<T> From<&HashMap<String, T>> for Item
where
    Item: From<HashMap<String, T>>,
    HashMap<String, T>: Clone,
{
    fn from(m: &HashMap<String, T>) -> Self {
        Item::from(m.clone())
    }
}
```

Reply:
That would definitely be better 👍 

I suggested to the SDK team to provide a method for taking ownership of the HashMap as well, which would be better than cloning, and they added it to a to-do list.

Plus, now I'm thinking it would probably be better to just have the user clone the HashMap anyway since it's an expensive operation so should probably be explicit.

Maybe this PR should just update the docs to use the method instead of the field, and then clone?

Reply:
> I suggested to the SDK team to provide a method for taking ownership of the HashMap as well, which would be better than cloning, and they added it to a to-do list.

Fantastic! Thank you.

Reply:
> Plus, now I'm thinking it would probably be better to just have the user clone the HashMap anyway since it's an expensive operation so should probably be explicit.
> 
> Maybe this PR should just update the docs to use the method instead of the field, and then clone?

Part of me was thinking this, but I also don't want to just say "no do it this other way".